client/reader/update-notice/index.jsx

```diff
@@ -52,4 +84,5 @@ class UpdateNotice extends React.PureComponent {
 
 export default connect( state => ( {
 	cappedUnreadCount: getDocumentHeadCappedUnreadCount( state ),
+	state,
```

Comment:
that's a bit scary. now we depend on the whole state?

Reply:
Mostly scary because getCommentById is on the call chain here and that thing is pretty non performant. Hrm. Can't really get it off the chain though can we...

Reply:
have a better idea?

i could move all the logic out to `mapStateToProps`

Reply:
yea ill do that. it guards from wasted unnecessary rerenders